packages/teams-js/src/private/filesExtensionsFor3PStorageproviders.ts

```diff
@@ -0,0 +1,194 @@
+    sequence: number;
+    /** A Blob object representing the data of the file chunks. */
+    file: Blob;
+  }
+
+  /**
+   * Get drag-and-drop files using a callback.
+   *
+   * @param dragAndDropInput - Input for drag-and-drop action.
+   * @param callback - Callback function to handle the result.
+   */
+  export function getDragAndDropFilesHandler(
+    dragAndDropInput: string,
+    callback: (attachments: FilesFor3PApps[], error?: SdkError) => void,
+  ): void {
+    if (!callback) {
+      throw new Error('[getDragAndDropFiles] Callback cannot be null');
+    }
+
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+
+    if (!dragAndDropInput || dragAndDropInput === '') {
+      //condition changed
+      const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
+      callback([], invalidInput);
+      return;
+    }
+    getFilesDragAndDropViaCallback(dragAndDropInput, callback);
+  }
+
+  function getFilesDragAndDropViaCallback(
+    dragAndDropInput: string,
+    callback: (attachments: FilesFor3PApps[], error?: SdkError) => void,
+  ): void {
+    const files: FilesFor3PApps[] = [];
+    let helper: filesExtensionsFor3PStorageproviders.AttachmentListHelper = {
+      fileType: '',
+      assembleAttachment: [],
+    };
+
+    function handleGetDragAndDropFilesCallbackRequest(
+      fileResult: filesExtensionsFor3PStorageproviders.FileResult,
+    ): void {
+      if (callback) {
+        if (fileResult && fileResult.error) {
+          callback([], fileResult.error);
+        } else {
+          if (fileResult && fileResult.fileChunk) {
+            try {
+              const assemble: filesExtensionsFor3PStorageproviders.AssembleAttachment = decodeAttachment(
+                fileResult.fileChunk,
+                fileResult.fileType,
+              );
+              helper.assembleAttachment.push(assemble);
+
+              if (fileResult.fileChunk.chunkSequence == Number.MAX_SAFE_INTEGER) {
+                const fileBlob = createFile(helper.assembleAttachment, fileResult.fileType);
+
+                if (fileResult.isLastFile) {
+                  // conver blob to File
+                  const receivedFile = new File([fileBlob], fileResult.fileName, {
+                    type: fileBlob.type,
+                  });
+
+                  files.push(receivedFile);
+
+                  callback(files, fileResult.error);
+                }
+
+                helper = {
+                  fileType: '',
+                  assembleAttachment: [],
+                };
+              }
+            } catch (e) {
+              callback([], { errorCode: ErrorCode.INTERNAL_ERROR, message: e });
+            }
+          } else {
+            callback([], { errorCode: ErrorCode.INTERNAL_ERROR, message: 'data received is null' });
+          }
+        }
+      }
+    }
+    sendMessageToParent('getDragAndDropFiles', [dragAndDropInput], handleGetDragAndDropFilesCallbackRequest);
```

Comment:
Please use capability name as prefix followed by the function name. For example, `filesExtensionsFor3PStorageproviders.getDragAndDropFiles`.

Reply:
Done